build/package/Microsoft.DotNet.Cli.Installer.PKG.targets

```diff
@@ -1,10 +1,132 @@
+      </PropertyGroup>
+
+      <ItemGroup>
+        <DistributionTemplateReplacement Include="{SharedFxComponentId}">
+          <ReplacementString>$(SharedFrameworkComponentId)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{SharedHostComponentId}">
+          <ReplacementString>$(SharedHostComponentId)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{HostFxrComponentId}">
+          <ReplacementString>$(HostFxrComponentId)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{CLISdkComponentId}">
+          <ReplacementString>$(SdkComponentId)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{CLISdkNugetVersion}">
+          <ReplacementString>$(SdkVersion)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{CLISdkBrandName}">
+          <ReplacementString>$(SdkBrandName)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{SharedFxBrandName}">
+          <ReplacementString>$(SharedFrameworkBrandName)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{SharedHostBrandName}">
+          <ReplacementString>$(SharedHostBrandName)</ReplacementString>
+        </DistributionTemplateReplacement>
+        <DistributionTemplateReplacement Include="{HostFxrBrandName}">
+          <ReplacementString>$(HostFxrBrandName)</ReplacementString>
+        </DistributionTemplateReplacement>
+      </ItemGroup>
+
+      <!-- Consumed By Publish -->
+      <ItemGroup>
+        <GeneratedInstallers Include="$(SdkInstallerFile);$(CombinedFrameworkSdkHostInstallerFile)" />
+      </ItemGroup>
     </Target>
 
+    <Target Name="GenerateSdkPkg" 
+            Inputs="$(SdkLayoutOutputDirectory);"
+            Outputs="$(SdkInstallerFile)"
+            DependsOnTargets="Init;Layout;SetupPkgInputsOutputs"
+            Condition=" '%(OSName)' == 'osx' ">
+      
+      <Exec Command="pkgbuild
+                     --root '$(SdkLayoutOutputDirectory)'
+                     --identifier '$(SdkComponentId)'
+                     --version '$(SdkVersion)'
+                     --install-location '$(PkgInstallDirectory)'
+                     --scripts '$(SdkPkgScriptsDirectory)'
+                     '$(SdkInstallerFile)'" />
+    </Target>
+
+    <Target Name="GenerateSdkProductArchive"
+            Inputs="$(SdkInstallerFile);
+                    $(DownloadedSharedFrameworkInstallerFile);
+                    $(DownloadedHostFxrInstallerFile);
+                    $(DownloadedSharedHostInstallerFile)"
+            DependsOnTargets="Init;Layout;SetupPkgInputsOutputs;GenerateSdkPkg"
```

Comment:
Yeah I get what you're saying, I'll shore it up some. One thing I want to make sure stays enabled is that you can run any one of these targets independently (for debugging)